tape/src/main/java/com/squareup/tape2/ObjectQueue.java

```diff
@@ -25,9 +26,6 @@
     return new InMemoryObjectQueue<>();
   }
 
-  /** The underlying {@link QueueFile} backing this queue, or null if it's only in memory. */
-  public abstract @Nullable QueueFile file();
```

Comment:
with `file():QueueFile` in `ObjectQueue` you would need to null-check on the result to be save, now with the method in `FileObjectQueue` you have to check for the implementation and are then guaranteed to get a `QueueFile` back, not a huge difference.
I think this could be improved by making the factory methods return the specific implementations.